typescript/utils/src/altvm.ts

```diff
@@ -0,0 +1,657 @@
+  WEIGHTED_MESSAGE_ID_MULTISIG = 'weightedMessageIdMultisigIsm',
+  CCIP = 'ccipIsm',
+  OFFCHAIN_LOOKUP = 'offchainLookupIsm',
+}
+
+export type ReqGetIsmType = { ismId: string };
+
+export type ReqMessageIdMultisigIsm = { ismId: string };
+export type ResMessageIdMultisigIsm = {
+  address: string;
+  threshold: number;
+  validators: string[];
+};
+
+export type ReqMerkleRootMultisigIsm = { ismId: string };
+export type ResMerkleRootMultisigIsm = {
+  address: string;
+  threshold: number;
+  validators: string[];
+};
+
+export type ReqRoutingIsm = { ismId: string };
+export type ResRoutingIsm = {
+  address: string;
+  owner: string;
+  routes: {
+    domainId: number;
+    ismId: string;
+  }[];
+};
+
+export type ReqNoopIsm = { ismId: string };
+export type ResNoopIsm = {
+  address: string;
+};
+
+export enum HookType {
+  CUSTOM = 'custom',
+  MERKLE_TREE = 'merkleTreeHook',
+  INTERCHAIN_GAS_PAYMASTER = 'interchainGasPaymaster',
+  AGGREGATION = 'aggregationHook',
+  PROTOCOL_FEE = 'protocolFee',
+  OP_STACK = 'opStackHook',
+  ROUTING = 'domainRoutingHook',
+  FALLBACK_ROUTING = 'fallbackRoutingHook',
+  AMOUNT_ROUTING = 'amountRoutingHook',
+  PAUSABLE = 'pausableHook',
+  ARB_L2_TO_L1 = 'arbL2ToL1Hook',
+  MAILBOX_DEFAULT = 'defaultHook',
+  CCIP = 'ccipHook',
+}
+
+export type ReqGetHookType = { hookId: string };
+
+export type ReqGetInterchainGasPaymasterHook = { hookId: string };
+export type ResGetInterchainGasPaymasterHook = {
+  address: string;
+  owner: string;
+  destinationGasConfigs: {
+    [domainId: string]: {
+      gasOracle: {
+        tokenExchangeRate: string;
+        gasPrice: string;
+      };
+      gasOverhead: string;
+    };
+  };
+};
+
+export type ReqGetMerkleTreeHook = { hookId: string };
+export type ResGetMerkleTreeHook = {
+  address: string;
+};
+
+// ### QUERY WARP ###
+
+export enum TokenType {
+  COLLATERAL = 'COLLATERAL',
+  SYNTHETIC = 'SYNTHETIC',
+}
+
+export type ReqGetToken = { tokenId: string };
+export type ResGetToken = {
+  address: string;
+  owner: string;
+  tokenType: TokenType;
+  mailboxId: string;
+  ismId: string;
+  originDenom: string;
+  name: string;
+  symbol: string;
+  description: string;
+  divisibility: number;
+};
+
+export type ReqGetRemoteRouters = { tokenId: string };
+export type ResGetRemoteRouters = {
+  address: string;
+  remoteRouters: {
+    receiverDomainId: number;
+    receiverContract: string;
+    gas: string;
+  }[];
+};
+
+export type ReqGetBridgedSupply = { tokenId: string };
+
+export type ReqQuoteRemoteTransfer = {
+  tokenId: string;
+  destinationDomainId: number;
+  customHookId: string;
+  customHookMetadata: string;
+};
+export type ResQuoteRemoteTransfer = { denom: string; amount: bigint };
+
+// ### POPULATE CORE ###
+
+export type ReqCreateMailbox = {
+  signer: string;
+  domainId: number;
+  defaultIsmId: string;
+};
+export type ResCreateMailbox = { mailboxId: string };
+
+export type ReqSetDefaultIsm = {
+  signer: string;
```

Comment:
i see that this types define the signer field but it is then ignored in the external API interface, should really define it if it is a internal implementation detail?

Reply:
It is only ignored in the Signer interface since we can infer the signer address from the private key. In the provider interface however when we populate (or create) the transactions we need it